src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -1620,56 +1620,83 @@ bool ExecutorWorker::deleteFinishedAsyncJobs() {
     return !hasError;
 }
 
-bool ExecutorWorker::isManagedBackError(const ExitCause exitCause, bool &isInconsistencyIssue) {
-    isInconsistencyIssue = exitCause == ExitCauseInvalidName;
-    return exitCause == ExitCauseInvalidName || exitCause == ExitCauseUploadNotTerminated || exitCause == ExitCauseApiErr ||
-           exitCause == ExitCauseFileTooBig || exitCause == ExitCauseNotFound;
+bool ExecutorWorker::handleManagedBackError(ExitCause jobExitCause, SyncOpPtr syncOp, bool isInconsistencyIssue) {
+    _executorExitCode = ExitCodeOk;
+    if (jobExitCause == ExitCauseQuotaExceeded) {
+        _syncPal->pause();
+    } else {
+        // The item should be temporarily blacklisted
+        _syncPal->blacklistTemporarily(syncOp->affectedNode()->id() ? *syncOp->affectedNode()->id() : std::string(),
+                                       syncOp->affectedNode()->getPath(),
+                                       syncOp->targetSide() == ReplicaSideLocal ? ReplicaSideRemote : ReplicaSideLocal);
+    }
+
+    affectedUpdateTree(syncOp)->deleteNode(syncOp->affectedNode());
+    if (syncOp->correspondingNode()) {
+        targetUpdateTree(syncOp)->deleteNode(syncOp->correspondingNode());
+    }
+
+    NodeId locaNodeId;
+    NodeId remoteNodeId;
+    if (syncOp->targetSide() == ReplicaSideLocal) {
+        if (syncOp->correspondingNode() && syncOp->correspondingNode()->id()) locaNodeId = *syncOp->correspondingNode()->id();
+        remoteNodeId = syncOp->affectedNode()->id() ? *syncOp->affectedNode()->id() : std::string();
+    } else {
+        locaNodeId = syncOp->affectedNode()->id() ? *syncOp->affectedNode()->id() : std::string();
+        if (syncOp->correspondingNode() && syncOp->correspondingNode()->id()) remoteNodeId = *syncOp->correspondingNode()->id();
+    }
+
+    Error error;
+    if (isInconsistencyIssue) {
+        error = Error(_syncPal->syncDbId(), locaNodeId, remoteNodeId, syncOp->affectedNode()->type(),
+                      syncOp->affectedNode()->getPath(), ConflictTypeNone, InconsistencyTypeForbiddenChar);
+    } else {
+        error = Error(_syncPal->syncDbId(), locaNodeId, remoteNodeId, syncOp->affectedNode()->type(),
+                      syncOp->affectedNode()->getPath(), ConflictTypeNone, InconsistencyTypeNone, CancelTypeNone, "",
+                      ExitCodeBackError, jobExitCause);
+    }
+    _syncPal->addError(error);
+
+    return true;
 }
 
+namespace details {
+bool isManagedBackError(ExitCause exitCause) {
+    static const std::set<ExitCause> managedExitCauses = {ExitCauseInvalidName,   ExitCauseApiErr,
+                                                          ExitCauseFileTooBig,    ExitCauseNotFound,
+                                                          ExitCauseQuotaExceeded, ExitCauseUploadNotTerminated};
+
+    return managedExitCauses.find(exitCause) != managedExitCauses.cend();
+}
+}  // namespace details
```

Comment:
Thanks for asking! I didn't do it (I actually did, then I reverted back to two maps) because the two maps are different and I don't know if they should coincide or not (afraid of unforeseen side effects).